kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -350,82 +394,60 @@ class KinesisSink private (
   private def putToSqs(sqs: SqsClientAndName, batch: List[Event]): Future[Unit] =
     Future {
       log.info(s"Writing ${batch.size} messages to SQS queue: ${sqs.sqsBufferName}")
-      val sqsBatchEntries = batch.map(addUuids).map(toSqsBatchEntry)
       val MaxSqsBatchSize = 10
-      val MaxRetries = 10
-      sqsBatchEntries.grouped(MaxSqsBatchSize).foreach { batchEntryGroup =>
-        sendToSqsWithRetry(sqs, batchEntryGroup, MaxRetries).transform {
-          case failure @ Failure(ex) =>
-            log.info(s"Sending to sqs failed with exception: $ex")
-            failure
-          case notSent @ Success(None) =>
-            log.info(s"Sending to sqs didn't succeed after $MaxRetries retries. Dropping events.")
-            notSent
-          case s @ Success(Some(_)) => s
+      batch
+        .map(toSqsBatchEntry)
+        .grouped(MaxSqsBatchSize)
+        .foreach { batchEntryGroup =>
+          sendToSqs(sqs, batchEntryGroup).transform {
+            case failure @ Failure(ex) =>
+              log.info(s"Sending to sqs failed with exception: $ex")
+              failure
+            case s @ Success(_) => s
```

Comment:
Why do we have return values if `putToSqs(sqs: SqsClientAndName, batch: List[Event])` returns `Future[Unit]` ?

Reply:
The cases are inside `.transform` function. The `Unit` comes from `foreach` in 401.